include/robot_interfaces/sensor_frontend.hpp

```diff
@@ -0,0 +1,58 @@
+#pragma once
+
+#include <algorithm>
+#include <cmath>
+
+#include <time_series/time_series.hpp>
+
+#include <robot_interfaces/sensor_data.hpp>
+
+namespace robot_interfaces
+{
+template <typename Type>
+using Timeseries = time_series::TimeSeries<Type>;
+typedef time_series::Index TimeIndex;
+
+/**
+ * @brief Communication link between SensorData and the user.
+ *
+ * Exposes the sensor data to the user via hardware-friendly methods.
```

Comment:
What do you mean by "hardware-friendly"?

Reply:
just to emphasize the point of the frontend (that is the methods the users can use won't break anything)

Reply:
The frontend does not really do anything from preventing the user to break the hardware, it just ensures they do not mess with the data in the time series.  The safety checks are happening in the backend (but anyway not for the sensor, as we don't send commands to it).